exercises/acronym/acronym_test.py

```diff
@@ -1,26 +1,40 @@
 import unittest
+from ddt import ddt, data, unpack
 
 from acronym import abbreviate
 
 
 # Tests adapted from `problem-specifications//canonical-data.json` @ v1.3.0
 
-class AcronymTest(unittest.TestCase):
-    def test_basic(self):
-        self.assertEqual(abbreviate('Portable Network Graphics'), 'PNG')
-
-    def test_lowercase_words(self):
-        self.assertEqual(abbreviate('Ruby on Rails'), 'ROR')
+class AnnotatedCase(dict):
+    case_args = ('phrase', 'expected')
 
-    def test_punctuation(self):
-        self.assertEqual(abbreviate('First In, First Out'), 'FIFO')
+    def __init__(self, name, *args):
+        setattr(self, '__name__', name)
+        for k, v in zip(AnnotatedCase.case_args, args):
+            self[k] = v
 
-    def test_all_caps_words(self):
-        self.assertEqual(abbreviate('GNU Image Manipulation Program'), 'GIMP')
 
-    def test_punctuation_without_whitespace(self):
-        self.assertEqual(
-            abbreviate('Complementary metal-oxide semiconductor'), 'CMOS')
+@ddt
+class AcronymTest(unittest.TestCase):
+    @data(
+        AnnotatedCase('basic', "Portable Network Graphics", 'PNG'),
+        AnnotatedCase('lowercase words', 'Ruby on Rails', 'ROR'),
+        AnnotatedCase('punctuation', 'First In, First Out', 'FIFO'),
+        AnnotatedCase(
+            'all caps word',
+            'GNU Image Manipulation Program',
+            'GIMP'
+        ),
+        AnnotatedCase(
+            'punctuation without whitespace',
+            'Complementary metal-oxide semiconductor',
+            'CMOS'
+        ),
+    )
+    @unpack
+    def test_abbreviate(self, phrase, expected):
+        self.assertEqual(abbreviate(phrase), expected)
```

Comment:
Interesting note: this test file actually is actually longer than the original. Clearly test suite size will only reduce where repetitive test code is either several lines long or repeated in more than just a few instances.